Automattic/jetpack
# Add Jetpack for Agencies module to dashboard to advertise agency dashboard

#### Changes proposed in this Pull Request:
Add a module to the Jetpack dashboard to advertise Jetpack for Agencies to users who are managing 2 or more sites. This card will take them to the [Jetpack Agencies page](https://jetpack.com/for/agencies/). 

It was decided that, for now, we do not need to hide this from users who are already defined as agencies due to there not being many agencies yet and the amount of work that would go into implementing that. This can be read about here: pbNhbs-1SF-p2#comment-9555

#### Jetpack product discussion
P2: p6TEKc-5Wl-p2

#### Does this pull request change what data or activity we track or use?
Yes, this PR adds tracking for how many clicks the "Learn More" button gets on the new module, as well as how many clicks the "dismiss" button on the module gets

#### Testing instructions:
1. Check out this branch via `git switch add/jetpack-agencies-module-to-dashboard`
2. Start the Jetpack Plugin locally on Docker, or however you go about testing the plugin locally
3. Apply any Jetpack Plan to your local wordpress site if you don't already have one active
4. Go to the Jetpack Dashboard at `/wp-admin/admin.php?page=jetpack#/dashboard`
5. Make sure you are on the `At a Glance` tab and scroll down until you see the new Agency module
![Screen Shot 2022-07-22 at 2 24 14 PM](https://user-images.githubusercontent.com/65001528/180543631-6a14d562-0451-4b88-8a28-fcf28b9f3929.jpg)
6. Switch to the `My Plan` tab and scroll down to make sure the module is still visible
7. Click on the `Learn More` CTA and ensure that this button leads to the [Jetpack Agencies page](https://jetpack.com/for/agencies)
8. Navigate to this file: 2cf70-pb and go to line 556 and change the number `2` to a number greater than the number of sites you manage. This section is checking to make sure the user is managing 2 or more sites, if you increase that number to a number greater than the number of sites you manage, the module should disappear (this is to test the logic behind conditionally showing the module based on site_count)
![image](https://user-images.githubusercontent.com/65001528/180558835-9e86fb69-bdda-4392-8fe4-94a1adabe5ef.png)
9. Lastly, click the `dismiss` button in the top right hand corner of the module. Doing so should make the module disappear. Refresh the page and ensure the module does not reappear after a refresh. 

https://user-images.githubusercontent.com/65001528/180561262-22b8e3ef-93bf-4a1f-bd68-f4506dda5b82.mov

NOTE: If you need to get the agencies module back after dismissing it, the easiest way I've found to do that is to navigate to [this line of code](https://github.com/Automattic/jetpack/pull/25041/files#diff-30fa007f36904a42a4886e3ce92058f8c3d9b155fe46f13246ca382c83eb76e1R106) and changing the boolean value to `false` and then calling the `dismiss()` function at the start of the `AgenciesCard` functional component. Save and then refresh the page and the module should be back (but make sure to revert this code before continued testing or the dismiss button won't work)
